model/src/main/kotlin/config/PackageConfiguration.kt

```diff
@@ -66,11 +67,19 @@ data class PackageConfiguration(
      * License finding curations.
      */
     @JsonInclude(JsonInclude.Include.NON_EMPTY)
-    val licenseFindingCurations: List<LicenseFindingCuration> = emptyList()
+    val licenseFindingCurations: List<LicenseFindingCuration> = emptyList(),
+
+    /**
+     * The source code origin this configuration applies to.
+     */
+    @JsonInclude(JsonInclude.Include.NON_NULL)
+    val sourceCodeOrigin: SourceCodeOrigin? = null
 ) {
     init {
-        require((sourceArtifactUrl == null) xor (vcs == null)) {
-            "A package configuration must either set the 'sourceArtifactUrl' or the 'vcs' property."
+        require(
+            listOfNotNull(sourceArtifactUrl, vcs, sourceCodeOrigin).size <= 1
```

Comment:
Would it make sense to make make `<=` be a `==` in this commit? And then make a commit on top, which makes `provenance optional? See also comment on the `matches()` function.